### PR TITLE
moved doc root to landing page, make user landing a guide page

### PR DESCRIPTION
**Pull request by story645**
Makes doc/index.rst the root of the doc tree, per  #26156 and discussion w/ @tacaswell. I took the approach of treating it as a sorta guide to the docs, which is why there's a little preamble before each section. It irks me that the section headers aren't all in the same case, but I'm also looking for something easily right hand parseable. #26328 was opened with this in mind to give a cleaner listing for the plot type gallery. 

Moving root to doc/index.rst means that user/index.rst is no longer root. Here I got rid of it and am using user_explain/index as the landing page for the user guide. Added anatomy + shortened explanation as mentioned in #26205 as a way to orient new users to the contents of the user guide. 

Changes here bring the docs more in line w/ the structure proposed in https://github.com/matplotlib/mpl-sphinx-theme/pull/72

attn: @esibinga and @melissawm

### CODE REVIEW

File: galleries/users_explain/index.rst [hunk 1]
```diff
@@ -31,7 +61,7 @@ Using Matplotlib
 
         .. toctree::
             :maxdepth: 2
-            :includehidden:
+            :hidden:
```
Comment (jklymak) [inline galleries/users_explain/index.rst:64]: Why are you hiding this?

File: galleries/users_explain/index.rst [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
 Using Matplotlib
-----------------
+================
+
+
+.. grid:: 1 1 2 2
+
+   .. grid-item::
+
+      .. image:: ../../_static/anatomy.png
```
Comment (jklymak) [inline galleries/users_explain/index.rst:17]: I still object to this, as in https://github.com/matplotlib/matplotlib/issues/26205 as necessary use of viewer real estate.

File: doc/devel/index.rst [hunk 3]
```diff
@@ -9,6 +9,9 @@ ways to contribute: optimizing and refactoring code, detailing unclear
 documentation and writing new examples, reporting and fixing bugs and requesting
 and implementing new features, helping the community...
 
+New contributors
```
Comment (jklymak) [inline doc/devel/index.rst:12]: All of these changes seem orthogonal to this PR...

File: doc/index.rst [hunk 4]
```diff
@@ -9,100 +7,155 @@
 Matplotlib |release| documentation
 ##################################
```
Comment (jklymak) [inline doc/index.rst:9]: The changes to this page mean I need to scroll down two screens on my pretty large monitor to see any useful information.  I preferred the old version - this is drifting back to the old version with too much info on the front page.  The current version also only came in 3.5, I think it's not nice to change the landing page too often.

File: doc/conf.py [hunk 5]
```diff
@@ -311,7 +311,7 @@ def gallery_image_warning_filter(record):
 source_encoding = "utf-8"
 
 # The toplevel toctree document (renamed to root_doc in Sphinx 4.0)
-root_doc = master_doc = 'users/index'
+root_doc = master_doc = 'index'
```
Comment (jklymak) [inline doc/conf.py:314]: users/index no longer links in the top level navigation.

File: doc/index.rst [hunk 6]
```diff
@@ -9,100 +7,164 @@
...
+            - Make figures: `.pyplot.figure`
+
+            - Make subplots: `.pyplot.subplots`, `.pyplot.subplot_mosaic`
+
+
+        .. grid-item::
+
+            - Figure methods: :doc:`Figure API <api/figure_api>`
+
+            - Plotting methods: :doc:`Axes API <api/axes_api>`
```
Comment (tacaswell) [inline doc/index.rst:95]: I do no think this add enough to the front page.

```suggestion
```

**Comment by tacaswell** [comment]
I agree with Jody, the content of the front page should stay about the same.   I did not mean to open a discussion about radically changing what was on the front page (sorry if I communicated badly). 

By making that the top of the TOC I think it would smooth out a bunch of the weirdness in how we are trying to use sphinx.  Can we please keep this PR to primarily doing that rotation and separate it from any drastic content changes?

**Comment by esibinga** [comment]
From a UX perspective, the small changes made to the **Learn** section make a big difference to usability with very little visual disruption to the existing interface. Adding a few more words and differentiating `How to use Matplotlib?` and `What can Matplotlib do?` ([here](https://output.circle-artifacts.com/output/job/865255a4-9010-4710-b26c-7b83011536aa/artifacts/0/doc/build/html/index.html)) set expectations really well so that new (and experienced) users can better see/intuit where they will find explanations and guidance vs. examples and demonstrations of mpl's visual capabilities.  Much more user-friendly IMO, and aligns with the structure and definitions we are working towards in the gallery tagging project.

**Review by jklymak — COMMENTED** [review]
The reorganization makes more sense than what we had before.  Glad it was relatively straight forward.  I'm largely fine with the output in https://output.circle-artifacts.com/output/job/865255a4-9010-4710-b26c-7b83011536aa/artifacts/0/doc/build/html/index.html

The current version doesn't render and needs a rebase?  so not 100% sure.

Other than the two major comments below, my other suggestion is that the new page is somewhat eclectic mix of cards and not cards - I can't see the reason for what is in a card and what is not in a card, so it makes it a bit confusing to my eye.

File: doc/index.rst [hunk 7]
```diff
@@ -9,100 +7,168 @@
...
+                .. code-block:: bash
+
+                    conda install -c conda-forge matplotlib
+
+        **Next Step**: :ref:`quick_start`
+
     .. grid-item::
 
-        Install using `pip <https://pypi.org/project/matplotlib>`__:
+        .. toctree::
```
[the remark this answers is not in the document]
Comment (jklymak) [reply doc/index.rst:39]: For sure everything needs to show up in a TOC tree. I guess we should be careful not to break that in this change.

**Comment by jklymak** [comment]
This would ideally be finished before 3.8 doc release.

File: doc/index.rst [hunk 8]
```diff
@@ -9,100 +7,176 @@
...
+    .. grid-item::
+        New versions are generally released every 6-8 months.
+        New features and API changes are described in the release notes.
+
+    .. grid-item::
 
-Matplotlib is a community project maintained for and by its users.  There are many ways
-you can help!
+        .. toctree::
+            :maxdepth: 2
```
Comment (timhoffm) [inline doc/index.rst:180]:
```suggestion
            :maxdepth: 1
```

The toc looks a bit too large compared to the description block on the left, and I don't think direct click-through links are needed here. Checking release notes is not something one does often. And anyway it may be simpler to click the "Releases" navbar item rather than scrolling down. So this is here manily for structure and as an entry point but not so much for quick navigation.

image

File: doc/index.rst [hunk 9]
```diff
@@ -9,100 +7,176 @@
...
+=====
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+
+        .. raw:: html
+
+            <div class="grid__intro" id="image_rotator"></div>
```
Comment (timhoffm) [inline doc/index.rst:139]: This feels out of place. What does this have to do with "About"?

If you want to have something visual, I suggest to either use the matplotlib logo, or if that's too colorful here, some sort of "about icon".

**Review by timhoffm — COMMENTED** [review]
I like the new front page very much. And it's a relieve that the awkward document structure gets sorted out. :rocket:

File: doc/index.rst (hunk 9 above)
Comment (timhoffm) [reply doc/index.rst:139]: > was thinking more plots are what the library is about,

Yes, but plots is not what this section is about. This section is a kind of meta-section about the library itself. There are dedicatedly all non-plotting aspects, which is why I find plots inappropriate here.

More generally/aesthetically: it's a bit imbalanced that this is the only non-text entry.

Comment (jklymak) [reply doc/index.rst:139]: I'm having a lot of trouble with the idea that the only plots on this page would be the old-timey plots above.  We got rid of those because they were visually unappealing, and I don't think anyone visiting the page is going to know enough history to understand we are making a historical reference.  

I think having a plot on this page is in general a good idea.  I'd actually put it at the top, next to Install, which doesn't need to be so wide.  

The "About" section could have a picture if you really want.  Perhaps made to look like a screen grab from an old browser or something.   But I'm not sure we need "About" on this page at all?  We do have a brochure site...

Comment (jklymak) [reply doc/index.rst]: > Don't want it next to Install 'cause it'll mess with finding it on the page. One of the things I've grown to really embrace is the 'on this page' menu - unfortunately I haven't yet figured out how to make headings in grids and cards show up there.

Just put the heading `Installation` outside the columns/grids?

Comment (jklymak) [reply doc/index.rst]: If you want it showing up in "On this page" then I guess it needs to be a heading?  I don't think we need it on this page, but...

**Comment by timhoffm** [comment]
Since #26414 is in, you should be able to rebase on master to get rid of the flake8 errors.

File: doc/index.rst (hunk 9 above)
Comment (jklymak) [reply doc/index.rst]: Your eeg example isn't in the right spot or didn't get committed to static.  

However I still really think it's a bad idea to put this plot on the top level of the docs as the only example of what we can do. I understand the logic, but it is a poor visualization, both practically and aesthetically.

Comment (jklymak) [reply doc/index.rst]: The advantage of the plot type images is they are all the same size and the same style. I think pulling random Gallery images would be hard, though if you had a curated list that may work. 

The plot type rotator is on the front page,  but I think would be useful here as well ( drives traffic to Plot Types and shows basic functionality). I would still have it near the top.

File: doc/index.rst [hunk 10]
```diff
@@ -9,100 +7,171 @@
...
+========
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+
+        Matplotlib was created by John Hunter to visualize EEG data. John's
+        goal was that Matplotlib make easy things easy and
+        hard things possible.
```
Comment (timhoffm) [inline doc/index.rst:170]: If you go into history, I suggest to span the arch to the current state

```suggestion
        Matplotlib was originally created by neurobiologist John Hunter to
        visualize EEG data. It quickly grew into a popular gerneral-purpose
        plotting library. John's goal was that Matplotlib make easy things
        easy and hard things possible.
```

---
On a side note, I still don't like the "easy things easy and hard things possible". The first part is to be expected from any reasonable tool. If you don't do this, you simply have bad user experience. IMHO that's not a mission statement worth mentioning. - It's in particular not great, as matplotlib has a relatively steep initial learning curve (partly due to our heritage) and I'd claim that e.g. plotly makes easy things easier than we do. :(

**Review by timhoffm — APPROVED** [review]
give or take my suggestions.
